webapp/sources/rudder/rudder-rest/src/test/scala/com/normation/rudder/MockServices.scala

```diff
@@ -2460,15 +2460,25 @@ class MockCampaign() {
         case Some(id) => afterDateFiltered.map(_.filter(_.start.isBefore(id)))
       }
 
+      def correctOrder(list : List[CampaignEvent]) =
+        asc match {
+          case Some("desc") => list.reverse
+          case _ => list
+        }
+      val ordered = order match {
+        case Some("startDate") => beforeDateFiltered.map(l => correctOrder(l.sortBy(_.start)))
```

Comment:
You can easily avoid one list pass by using `sortWith` and compare on start / end date

Reply:
I confess that i did not overthink the test implementation :)